src/ServiceResolver/ActualDefinition.php

```diff
@@ -31,11 +31,11 @@ public static function build(mixed $from): self
 
         return match (is_callable($from)) {
```

Comment:
I've found most of the `match` constructs hard to read. Maybe some of them would be more legible as `if (X) A else B` instead of `match (X) true => A; false => B`. This applies for `null/default` matchers as well.

Reply:
I think all the confusing `match` statements were removed here 7277202.

I left some that I find good usages and it can be also a try to see if we get used with it :)